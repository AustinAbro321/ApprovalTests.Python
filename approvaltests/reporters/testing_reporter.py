from approvaltests.core.reporter import Reporter


class ReporterForTesting(Reporter):
    def __init__(self) -> None:
        self.called = False

    def report(self, received_path: str, approved_path: str) -> None:
        self.called = True
