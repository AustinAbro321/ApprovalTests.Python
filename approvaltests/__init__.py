__author__ = 'Chris Lucian; Llewellyn Falco; Jim Counts; Tim Preece'

from .utils import *
from .asserts import *
from .approvals import *
from .combination_approvals import *