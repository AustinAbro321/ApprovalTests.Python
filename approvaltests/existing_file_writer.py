import shutil

from approvaltests.core import Options
from approvaltests.core import Writer


class ExistingFileWriter(Writer):
    def __init__(self, file_name: str, options: Options) -> None:
        self.file_name = file_name
        self.options = options

    def write_received_file(self, received_file: str) -> str:
        if not self.options.has_scrubber():
            shutil.copyfile(self.file_name, received_file)
        else:
            with open(self.file_name, mode="r") as file:
                text = file.read()
            text = self.options.scrub(text)
            with open(received_file, mode="w") as file:
                file.write(text)
        return received_file
