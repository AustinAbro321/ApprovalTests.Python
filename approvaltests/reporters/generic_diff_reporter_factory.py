import json
from typing import Any, Iterator, Union, List

from approvaltests.reporters import Reporter
from approvaltests.reporters.generic_diff_reporter import GenericDiffReporter, GenericDiffReporterConfig, create_config
from approvaltests.utils import get_adjacent_file


class GenericDiffReporterFactory(object):
    reporters : List[Reporter] = []

    def __init__(self) -> None:
        self.load(get_adjacent_file("reporters.json"))

    def add_default_reporter_config(self, config):
        self.reporters.insert(0, config)
        self.reporters2.insert(0, create_config(config))

    def list(self) -> List[str]:
        return [r[0] for r in self.reporters]

    def list2(self) -> List[str]:
        return [r.name for r in self.reporters2]

    def get(self, reporter_name: str) -> GenericDiffReporter:
        config = next((r for r in self.reporters if r[0] == reporter_name), None)
        return self._create_reporter(config)

    def get2(self, reporter_name: str) -> Union[GenericDiffReporter, None]:
        config = next((r for r in self.reporters2 if r.name == reporter_name), None)
        if not config:
            return None
        return self._create_reporter2(config)

    @staticmethod
    def _create_reporter(config: Union[List[str], List[Union[str, List[str]]]]) -> GenericDiffReporter:
        if not config:
            return None
        return GenericDiffReporter(config)

    @staticmethod
    def _create_reporter2(config: GenericDiffReporterConfig) -> GenericDiffReporter:
        return GenericDiffReporter(config)

    def save(self, file_name: str) -> str:
        with open(file_name, "w") as f:
            json.dump(
                self.reporters, f, sort_keys=True, indent=2, separators=(",", ": ")
            )
        return file_name

    def save2(self, file_name: str) -> str:
        with open(file_name, "w") as f:
            json.dump(
                [reporter.serialize() for reporter in self.reporters2], f, sort_keys=True, indent=2, separators=(",", ": ")
            )
        return file_name

    def load(self, file_name: str) -> List[Union[List[str], List[Union[str, List[str]]]]]:
        with open(file_name, "r") as f:
            self.reporters = json.load(f)
        return self.reporters

    def load2(self, file_name: str) -> List[GenericDiffReporterConfig]:
        with open(file_name, "r") as f:
            configs = json.load(f)
        self.reporters2 = [create_config(config) for config in configs]
        return self.reporters2

    def get_first_working(self) -> GenericDiffReporter:
        working = (i for i in self.get_all_reporters() if i.is_working())
        return next(working, None)

    def get_all_reporters(self) -> Iterator[Any]:
        instances = (self._create_reporter(r) for r in self.reporters)
        return instances

    def get_all_reporters2(self) -> Iterator[GenericDiffReporter]:
        instances = (self._create_reporter2(r) for r in self.reporters2)
        return instances

    def remove(self, reporter_name: str) -> None:
        self.reporters = [r for r in self.reporters if r[0] != reporter_name]

    def remove2(self, reporter_name: str) -> None:
        self.reporters2 = [r for r in self.reporters2 if r.name != reporter_name]
