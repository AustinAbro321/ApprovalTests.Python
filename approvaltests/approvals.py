import json
from threading import local

from approvaltests.approval_exception import ApprovalException
from approvaltests.core.namer import Namer
from approvaltests.file_approver import FileApprover
from approvaltests.reporters.diff_reporter import DiffReporter
from approvaltests.string_writer import StringWriter

DEFAULT_REPORTER = local()


def set_default_reporter(reporter):
    global DEFAULT_REPORTER
    DEFAULT_REPORTER.v = reporter
    

def get_default_reporter():
    if not hasattr(DEFAULT_REPORTER, "v") or DEFAULT_REPORTER.v is None:
        return DiffReporter()
    return DEFAULT_REPORTER.v


def verify(data, reporter=None):
    verify_with_namer(data, get_default_namer(), reporter)


def verify_with_namer(data, namer, reporter):
    reporter = get_reporter(reporter)
    approver = FileApprover()
    writer = StringWriter(data)
    error = approver.verify(namer, writer, reporter)
    if error is not None:
        raise ApprovalException(error)


def get_reporter(reporter):
    if reporter is None:
        reporter = get_default_reporter()
    return reporter


def get_default_namer():
    return Namer()


def verify_all(header, alist, formatter=None, reporter=None):
    if formatter is None:
        formatter = PrintList().print_item 
    text = header + '\n\n'
    for i in alist:
        text += formatter(i) + '\n'
    verify(text, reporter)


def verify_as_json(object, reporter=None):
    n_ = to_json(object) + "\n"
    verify(n_, reporter)

def to_json(object):
    return json.dumps(
        object,
        sort_keys=True,
        indent=4,
        separators=(',', ': '),
        default=lambda o: o.__dict__)


class PrintList(object):
    index = 0

    @classmethod
    def print_item(cls, x):
        text = str(cls.index) + ') ' + str(x)
        cls.index += 1
        return text


def verify_file(file_name, reporter=None):
    with open(file_name, 'r') as f:
        file_contents = f.read()
        verify(file_contents, reporter)


