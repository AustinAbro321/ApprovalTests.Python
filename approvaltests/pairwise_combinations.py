from itertools import product
from typing import Any, Callable, Optional, Tuple, Union, List, Sequence

from approvaltests import (
    verify_with_namer,
    get_default_namer,
    Reporter,
    initialize_options,
    Options,
    verify,
)
from approvaltests.core.namer import StackFrameNamer
from approvaltests.reporters.testing_reporter import ReporterForTesting



def get_best_covering_pairs(
    input_arguments: Sequence[Sequence[Any]],
) -> Sequence[Sequence[Any]]:

    p = product(*input_arguments)
    return list(p)#[20:]
