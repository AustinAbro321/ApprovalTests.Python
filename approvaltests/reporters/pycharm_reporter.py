import json

from approvaltests.reporters.generic_diff_reporter import GenericDiffReporter


class PyCharmReporter(GenericDiffReporter):
    def __init__(self, path=None):
        config = ['PyCharm', path or "/Applications/PyCharm CE.app/Contents/MacOS/pycharm"]
        GenericDiffReporter.__init__(self, config)

    def get_command(self, received, approved):
        return [
            self.path,
            "diff",
            received,
            approved
        ]

    def __str__(self):
        return json.dumps(
            {
                'name': self.name,
                'path': self.path,
                'clazz': "PyCharmReporter"
            },
            indent=4,
            separators=(',', ': ')
        )
