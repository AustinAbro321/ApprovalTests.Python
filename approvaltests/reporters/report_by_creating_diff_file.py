import os
from difflib import unified_diff
from pathlib import Path

from approvaltests import ensure_file_exists
from approvaltests.reporters.generic_diff_reporter import GenericDiffReporter, GenericDiffReporterConfig


class ReportByCreatingDiffFile(GenericDiffReporter):
    def __init__(self):
        super().__init__(
            config=GenericDiffReporterConfig(
                name=self.__class__.__name__, path=r"C:\Windows\System32\fc.exe"
            )
        )

    def report(self, received_path: str, approved_path: str) -> bool:
        diff = self.calculate_diff(received_path,approved_path)
        suffix = Path(received_path).suffix
        diff_file_path = received_path.replace(f".received{suffix}", ".diff")
        print("diff file path text is ", diff_file_path)
        with open(diff_file_path, mode="w") as f1 :
            f1.write(diff)
        return True

    @staticmethod
    def calculate_diff(file1: str, file2: str):
        with open(file1) as f1:
            with open(file2) as f2:
                diff = unified_diff(
                    f2.readlines(),
                    f1.readlines(),
                    os.path.basename(file2),
                    os.path.basename(file1),
                )
                diff_string = "".join(diff)
                return diff_string

