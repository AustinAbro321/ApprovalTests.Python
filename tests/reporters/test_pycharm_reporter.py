
from approvaltests import verify

from approvaltests.reporters.pycharm_reporter import PyCharmReporter


def test_pycharm_diff_command():
    reporter = PyCharmReporter()
    received_path = "received.txt"
    approved_path = "approved.txt"
    verify(str(reporter.get_command(received_path, approved_path)))


def test_override_path():
    reporter = PyCharmReporter(path="foobar")
    received_path = "received.txt"
    approved_path = "approved.txt"
    verify(str(reporter.get_command(received_path, approved_path)))
