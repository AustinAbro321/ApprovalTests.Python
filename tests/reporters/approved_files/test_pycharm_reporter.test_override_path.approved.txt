['foobar', 'diff', 'received.txt', 'approved.txt']